compiler/rustc_codegen_ssa/src/traits/consts.rs

```diff
@@ -39,7 +51,102 @@ pub trait ConstCodegenMethods: BackendTypes {
 
     fn const_data_from_alloc(&self, alloc: ConstAllocation<'_>) -> Self::Value;
 
-    fn scalar_to_backend(&self, cv: Scalar, layout: abi::Scalar, llty: Self::Type) -> Self::Value;
+    fn const_bitcast(&self, val: Self::Value, ty: Self::Type) -> Self::Value;
+    fn const_pointercast(&self, val: Self::Value, ty: Self::Type) -> Self::Value;
+    fn const_int_to_ptr(&self, val: Self::Value, ty: Self::Type) -> Self::Value;
+    fn const_ptr_to_int(&self, val: Self::Value, ty: Self::Type) -> Self::Value;
+    /// Create a global constant.
+    ///
+    /// The returned global variable is a pointer in the default address space for globals.
+    fn static_addr_of_impl(&self, cv: Self::Value, align: Align, kind: Option<&str>)
```

Comment:
Maybe call this `static_addr_of_const`?

Reply:
ah fun, we only do that in some vtable paths, but not in the one in `scalar_to_backend`